test/Interop/Cxx/class/method/Inputs/methods.h

```diff
@@ -33,6 +33,8 @@ struct HasMethods {
 
   NonTrivialInWrapper nonConstPassThroughAsWrapper(int a) { return {a}; }
   NonTrivialInWrapper constPassThroughAsWrapper(int a) const { return {a}; }
+
+  void nonTrivialTakesRef(const NonTrivialInWrapper& w) const {}
```

Comment:
Does your change affect non-`const` references?

Reply:
Hmm, yes and no. Yes as in I did not add extra code to avoid adding `@_addressable` to non-const references. But no, because non-const references are imported as inout arguments so the compiler introducing copies for those arguments would be semantic breaking. We should not observe copies there even before this patch, adding the attribute should be noop. Would you prefer avoiding adding the attribute in that case?

Reply:
> non-const references are imported as inout arguments so the compiler introducing copies for those arguments would be semantic breaking

I thought the semantics of `inout` arguments (in pure Swift) were that they are copied in, and then later copied out? So doesn't that mean they _are_ copied, or am I misunderstanding how `inout` parameters work?

In any case, I get your point now about why we don't need to worry about non-const references for adding this attribute. I don't think you need to avoid adding it for non-const references. However, I do think it would be a good idea to add a test case for non-const references that checks (and documents) that scenario.

Reply:
> they are copied in, and then later copied out
I think the idea is that in pure Swift you cannot really observe copies so copying in and out is actually semantically the same as passing by address. This is not really true for C++, copies are observable through the copy ctor/assignment operations. The Swift compiler is doing best effort to avoid unnecessary copies for non-trivial C++ types whenever possible. 

> it would be a good idea to add a test case for non-const references 

Added a test.